### PR TITLE
Passing vetoes to scripts in pygrb results sub workflow

### PR DESCRIPTION
**Pull request by pannarale**
This PR is the second in the series started in PR #4929.

## Standard information about the request (and the following ones that will be linked to this)

This is a: a new feature enabling veto definer file usage in PyGRB.  Utilities and scripts in results production are being  streamlined along the way.

This change affects: PyGRB

This change changes: result presentation / plotting and scientific output.

If this change breaks the standard automated test running `--help` for PyGRB plotting scripts, I will add some workarounds to avoid this.  If needed, these will likely be empty functions: the plotting scripts will be progressively renovated in the whole series of PRs.

## Motivation
Prior to these PRs, PyGRB was not using data vetoes.

## Contents
- The results workflow generator for PyGRB is now passing segments files and the vdf to all results scripts
- The utility functions involved in loading trigger/injeciton data in pygrb were generalised
- `pycbc_pygrb_plot_skygrid` was updated accordingly as a first example.

## Testing performed
The totality of the changes that will be broken down in multiple PRs was tested on GRB 170817A data by producing a full results webpage (see [here](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_nov2024_1/)).

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: pycbc/results/pygrb_postprocessing_utils.py [hunk 1]
```diff
@@ -287,13 +287,21 @@ def _dataset_iterator(g, prefix=''):
 # =============================================================================
 # Functions to load triggers
 # =============================================================================
-def load_triggers(input_file, ifos, vetoes, rw_snr_threshold=None,
-                  slide_id=None):
+def _load_triggers(input_file, ifos, data_tag=None, rw_snr_threshold=None,
+                   slide_id=None):
     """Loads triggers from PyGRB output file, returning a dictionary"""
 
+    logging.info("Loading triggers.")
```
Comment (titodalcanton) [inline pycbc/results/pygrb_postprocessing_utils.py:294]: Kind of a silly point, but looking a few lines below, I wonder if this should say "Loading injections" if `data_tag` is "injs"?

File: pycbc/results/pygrb_postprocessing_utils.py [hunk 2]
```diff
@@ -336,12 +354,29 @@ def load_triggers(input_file, ifos, vetoes, rw_snr_threshold=None,
...
             if trigs_dict[path].size == trigs['network/slide_id'][:].size:
-                trigs_dict[path] = slide_filter(trigs, trigs_dict[path],
-                                                slide_id=slide_id)
+                trigs_dict[path] = _slide_filter(trigs, trigs_dict[path],
+                                                 slide_id=slide_id)
 
     return trigs_dict
 
 
+# Wrapper function to load trigger/injection data
```
Comment (titodalcanton) [inline pycbc/results/pygrb_postprocessing_utils.py:363]: Is it really necessary to introduce such a function? It seems you could just keep using `_load_triggers()`, and just add
```python
if not input_file:
    return None
```
at the very beginning.

File: bin/pygrb/pycbc_pygrb_results_workflow [hunk 3]
```diff
@@ -218,6 +220,12 @@ if not set(inj_sets).issubset(eff_secs):
 inj_sets = [i.upper() for i in inj_sets]
 inj_files = labels_in_files_metadata(inj_sets, start_rundir, args.inj_files)
 
+# File instance of the veto file
+veto_file = args.veto_file
+if veto_file: 
+    veto_file = os.path.join(start_rundir, args.veto_file)
```
Comment (titodalcanton) [inline bin/pygrb/pycbc_pygrb_results_workflow:226]: Why is this necessary? Should we not assume that the workflow generator will pass a complete file path to `--veto-file`?

Comment (titodalcanton) [reply bin/pygrb/pycbc_pygrb_results_workflow:226]: Also not sure you really need the `resolve_url_to_file()` call below…

Comment (titodalcanton) [reply bin/pygrb/pycbc_pygrb_results_workflow:226]: Ah wait, this is the results workflow generator. Then it makes more sense.

Comment (pannarale) [reply bin/pygrb/pycbc_pygrb_results_workflow:226]: Correct, this is workflow generation stuff, so the files are getting converted to `File` instances.

File: pycbc/results/pygrb_postprocessing_utils.py (hunk 2 above)
Comment (pannarale) [reply pycbc/results/pygrb_postprocessing_utils.py]: I blended the two functions together.
The idea of returning None was to avoid that check in all plotting scripts, which may or may not receive an `--inejection-file`.  I don't mind putting it there, i.e.,
```python
if args.injection_file:
    inj_data = load_data(...)
```

Comment (titodalcanton) [reply pycbc/results/pygrb_postprocessing_utils.py]: I see, then I think it makes sense.

**Review by titodalcanton — APPROVED** [review]
